src/Gemstone.Data/Model/TableOperations.cs

```diff
@@ -1322,15 +1790,56 @@ private T LoadRecordWithKeys(DataRow row)
     /// </summary>
     /// <param name="records">The collection of records to be inserted into the data table.</param>
     /// <returns>A data table containing data from the given records.</returns>
-    public DataTable ToDataTable(IEnumerable<T> records)
+    public DataTable ToDataTable(IEnumerable<T?> records)
+    {
+        DataTable dataTable = s_tableSchema.Clone();
+
+        foreach (T? record in records)
+        {
+            if (record is null)
+                continue;
+
+            DataRow row = dataTable.NewRow();
+
+            foreach (PropertyInfo property in s_properties.Values)
+                row[s_fieldNames[property.Name]] = property.GetValue(record);
+
+            dataTable.Rows.Add(row);
+        }
+
+        return dataTable;
+    }
+
+    DataTable ITableOperations.ToDataTable(IEnumerable records)
+    {
+        try
+        {
+            return ToDataTable(records.Cast<T?>());
+        }
+        catch (InvalidCastException ex)
+        {
+            throw new ArgumentException($"One of the provided records cannot be converted to type \"{typeof(T).Name}\": {ex.Message}", nameof(records), ex);
+        }
+    }
+
+    /// <summary>
+    /// Converts the given collection of <paramref name="records"/> into a <see cref="DataTable"/>.
+    /// </summary>
+    /// <param name="cancellationToken">Propagates notification that operations should be canceled.</param>
+    /// <param name="records">The collection of records to be inserted into the data table.</param>
+    /// <returns>A data table containing data from the given records.</returns>
+    public async Task<DataTable> ToDataTableAsync(IAsyncEnumerable<T?> records, CancellationToken cancellationToken)
     {
         DataTable dataTable = new(s_tableName);
 
         foreach (PropertyInfo property in s_properties.Values)
             dataTable.Columns.Add(new DataColumn(s_fieldNames[property.Name]));
```

Comment:
`ToDataTable()` is using `s_tableSchema.Clone()`. Should we use that here in `ToDataTableAsync()` as well?

Reply:
Probably, just for consistency. Adding...